quic/s2n-quic-transport/src/stream/stream_container.rs

```diff
@@ -555,19 +608,36 @@ impl<S: StreamTrait> StreamContainer<S> {
         );
     }
 
+    #[cfg(test)]
+    pub fn iterate_retransmission_list<F>(
+        &mut self,
+        controller: &mut stream::Controller,
+        mut func: F,
+    ) where
+        F: FnMut(&mut S) -> StreamContainerIterationResult,
+    {
+        iterate_interruptible!(
+            self,
+            waiting_for_retransmission,
+            waiting_for_retransmission_link,
+            controller,
+            func
+        );
+    }
+
     /// Iterates over all `Stream`s which are waiting for retransmission,
     /// and executes the given function on each `Stream`
     ///
     /// The `stream::Controller` will be notified of streams that have been
     /// closed to allow for further streams to be opened.
-    pub fn iterate_retransmission_list<F>(
+    pub fn send_on_retransmission_list<F>(
         &mut self,
         controller: &mut stream::Controller,
         mut func: F,
     ) where
         F: FnMut(&mut S) -> StreamContainerIterationResult,
     {
-        iterate_interruptible!(
+        send_on_transmission_list!(
```

Comment:
Does this have any weird interaction in the logic of the counters? Like if we go to retransmit and then call transmit on the same packet?

Reply:
Hmmm, I guess there is a bug here. To be honest, I'm not sure how we want batching to work with the retransmit list. Minimally, the retransmit and transmit should not share the same counter since there's no reason why their stream list would be the same (e.g. the head node of the retransmit list may be a different stream node than the transmit list).

Although I also don't know how important retransmission fairness is?